### PR TITLE
ROC-1105: Correctly set headers on inbound requests

### PR DESCRIPTION
**Pull request by rlewan**
Correctly propagates headers received on inbound requests as defined in [Reform logging specification](https://tools.hmcts.net/confluence/display/RTA/Logging).

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/logging/filters/RequestIdsSettingFilter.java [hunk 1]
```diff
@@ -35,38 +35,52 @@ public RequestIdsSettingFilter(Supplier<String> requestIdGenerator) {
...
         }
     }
 
+    private void setRequestTracingHeaders(HttpServletRequest httpServletRequest) {
+        String requestId = httpServletRequest.getHeader(HttpHeaders.REQUEST_ID);
+        if (isPresent(requestId)) {
+            MdcFields.setRequestId(requestId);
+            String rootRequestId = httpServletRequest.getHeader(HttpHeaders.ROOT_REQUEST_ID);
+            MdcFields.setRootRequestId(isPresent(rootRequestId) ? rootRequestId : requestId);
+            MdcFields.setOriginRequestId(httpServletRequest.getHeader(HttpHeaders.ORIGIN_REQUEST_ID));
```
Comment (timja-kainos) [inline src/main/java/uk/gov/hmcts/reform/logging/filters/RequestIdsSettingFilter.java]: is it fine to set null here if the origin request id doesn't exist?

File: src/main/java/uk/gov/hmcts/reform/logging/filters/RequestIdsSettingFilter.java [hunk 2]
```diff
@@ -35,38 +35,52 @@ public RequestIdsSettingFilter(Supplier<String> requestIdGenerator) {
...
         HttpSession session = httpServletRequest.getSession(false);
         return session == null ? null : session.getId();
     }
 
-    /**
-     * {@inheritDoc}.
-     */
+    private boolean isPresent(String string) {
+        return string != null && !string.isEmpty();
+    }
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/logging/filters/RequestIdsSettingFilter.java]: Is Guava or Apache Commons imported? if so, perhaps just `Strings.isNullOrEmpty` (or whatever it is ;) ) could be used?

Comment (rlewan) [reply src/main/java/uk/gov/hmcts/reform/logging/filters/RequestIdsSettingFilter.java]: It's not, I looked for it :( Do we want to bring it in just for this?

Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/logging/filters/RequestIdsSettingFilter.java]: in that case it's perfectly fine

**Pull request merged**